core/string/ustring.cpp

```diff
@@ -1644,6 +1644,17 @@ String String::hex_encode_buffer(const uint8_t *p_buffer, int p_len) {
 	return ret;
 }
 
+Vector<uint8_t> String::hex_decode() const {
+	const String *s = this;
+	Vector<uint8_t> out;
+	int len = s->size() / 2;
+	out.resize(len);
+	for (int i = 0; i < len; i++) {
+		out.write[i] = s->substr(i * 2, 2).hex_to_int();
```

Comment:
`hex_to_int()` has extra checks (sign, `0x` prefix, etc). I think it's undesirable to use it here.

Reply:
What should i use?

Reply:
You can modify the fragment from the `hex_to_int()` implementation.

https://github.com/godotengine/godot/blob/09b97662f9732161b153a0d008546ea4421a0e4b/core/string/ustring.cpp#L2171-L2179